test/server/services/schedule-service.test.js

```diff
@@ -0,0 +1,37 @@
+describe('Schedule service tests', () => {
+  let mockScheduleRespository = require('../repository/schedule-repository.mock')
+  let scheduleService
+
+  beforeEach(async () => {
+    jest.mock('../../../server/repository/schedule-repository', () => mockScheduleRespository)
+    scheduleService = require('../../../server/services/schedule-service')
+  })
+
+  afterEach(async () => {
+    jest.unmock('../../../server/repository/schedule-repository')
+  })
+
+  test('create creates all payment schedules', async () => {
+    const claim = {
+      claimId: 'MINE003'
+    }
+    const spy = jest.spyOn(mockScheduleRespository, 'create')
+
+    await scheduleService.create(claim)
+
+    expect(spy).toHaveBeenCalledTimes(6)

```

Comment:
Just testing the number of times called isn't enough to verify the code is working, the values returned need to be tested too.

Reply:
Agree, this is a general theme that the unit tests are a bit weak in this service.  I've created a separate card (379) to address the unit tests and get the code coverage up.  Usually I wouldn't want this to be merged without proper test coverage but given it's a POC and our impending priorities, I'd prefer not to hold this user story up on this occasion and pick it up through 379.

Thoughts?

Reply:
That's fine, though we need to ensure we're testing expectations as well as hitting the coverage.